File: 프로그래머스/3/12987. 숫자 게임/숫자 게임.js
function solution(A, B) {
    var answer = -1;
    A.sort((a, b) => a - b);
    B.sort((a, b) => a - b);
    let a = 0;
    let b = 0;
    answer = 0;
    while(b < B.length){
        const anum = A[a];
        const bnum = B[b];
        if(anum < bnum){
            answer ++;
            a++;
        }
        b++;
    }
    return answer;
}

File: 프로그래머스/3/12987. 숫자 게임/숫자 게임.java
import java.util.*;

class Solution {
    public int solution(int[] A, int[] B) {
        int answer = 0;
        Arrays.sort(A);
        Arrays.sort(B);
        
        int bp = 0; 
        int ap = 0;
        
        while(bp != B.length && ap != A.length){
            if(A[ap] >= B[bp]){
                bp++;
            }else{
                ap++;
                bp++;
                answer++;
            }
        }
        return answer;
    }
}